### DISCUSSION TITLE
Reorganize with a start.sh in the root folder and everything else in a /src or /lib folder

### DISCUSSION THREAD

**Discussion by synth-mania** [NimbleAINinja/Automated-AI-Web-Researcher-Hosted#4] [Ideas]
The idea is to give the project a more polished user experience by hiding internals within a subfolder, and also automating starting and disabling the venv and getting dependencies when running the project

I'll make a relevant pull request sometime today

**Comment by NimbleAINinja** [comment]
Please refresh before doing the PR, I just deployed a big update.

**Reply by synth-mania** [reply]
sounds good

**Reply by synth-mania** [reply]
I also am working on big changes.
They haven't been completed yet, and there are still some errors, but you can check them out:
https://github.com/synth-mania/NimbleAINinja-Auto-AI-Researcher

I hope we can resolve the merge conflicts and add some of these changes to your fork when I'm done.

I'm implementing a new config system that uses a config.json.
I also did the reorganization and added a start.sh

Already if you clone my fork and run start.sh, you should be brought to a configuration wizard.

Aside from that, there is more work to be done with integrating those changes properly with the rest of the codebase, but I'm curious what you think of my changes so far.

**Comment by NimbleAINinja** [comment]
Oh, I also put some config items in json.

Why would you build an interactive configuration thing? I kind of assume anyone capable of cloning a project from GitHub would feel comfortable ending a config directly in a text editor. 

This merge is going to be a ballache with the files in different folders. Also, having spent much time with this code today, I'm not so sure I want to continue working on it. We'll see.

**Reply by synth-mania** [reply]
I figured it would be nice to have, no other reason.

I agree, the merge is gonna be a pain. I'll probably just sync my fork with yours, resolve any conflicts, then submit a PR. You won't have to deal with the conflict then.

**Comment by synth-mania** [comment]
@NimbleAINinja I think you'll like this. I synced my fork with your most recent commits, and then redid the reorganization. I didn't reimplement the setup wizard or anything, all I did was create a start.sh, update the readme accordingly, and put all the internals into a subfolder. Please let me know what you think.
